lms/services/lti_outcomes.py

```diff
@@ -45,26 +45,20 @@ def read_result(self, lis_result_sourcedid):  # pylint:disable=no-self-use
             return None
 
     def record_result(  # pylint:disable=no-self-use
-        self, lis_result_sourcedid, score=None, **canvas_extras,
+        self, lis_result_sourcedid, score=None, pre_record_hook=None,
```

Comment:
It's a fuzzy area and both terms are used interchangeably (including by me in this PR apparently). Some SO answers:

 * https://stackoverflow.com/questions/11087543/what-is-the-difference-between-hook-and-callback#11087727

To the extent there is a difference I think it's something like:

 * Callbacks are called to inform them of an outcome (so that un-related processing can take place) for example "onSuccess"
 * Hooks are probably a super-set, but include the possibility of modify the process they are called in "preRecord" 
* There are some ancient interpretations of hooks which aren't really applicable in a Python world involving some sort of memory monkey patching to force a method to do something additional

As a result of hooks being a super-set of callbacks, and callbacks being much more common, I think that hooks are often thought of in terms of how they differ from callbacks (intended to modify behavior), but they technically don't imply that.

I'd therefore suggest the following:

 * If we want a term to only include modifying behavior "hook" is correct
 * If we want a term to only include return information "callback" is more specific (but "hook" is still correct)
 * If we want a term to encompass both, only "hook" is correct
 * It's a good idea to add something to a variable to let people know it's a function like `_fn` so it doesn't get mistaken for a regular value:
   * Basically a heads up: "functional programming ahead" to get in the right head space
   * This prevents the "why the hell are there parens on this?" moment
   * ... but we don't want to end-up in C land with `blah_bool`, `blah_str` etc.
 * As "hook" is more generally correct and shorter I'd suggest it's good practice to add it to the end

As for changes in this PR, I should definitely pick a side of the fence and be consistent. So I'll rename references to callbacks to hooks. It's also good to have this discussion because I realised I'd never really